websites/D/DscJobs/presence.ts

```diff
@@ -1,94 +1,114 @@
-      presenceData.details = "Creating Profile";
-      break;
-    case "/legal":
-      presenceData.details = "Viewing Legal Page";
-      break;
-    case "/partners":
-      presenceData.details = "Viewing Partners Page";
-      break;
-    case "/supporters":
-      presenceData.details = "Viewing Supporters Page";
-      break;
-    case "/settings":
-      presenceData.details = "Modifying CV";
-      break;
-    case "/profile":
-      presenceData.details = "Viewing Profile Page";
-      break;
-    case "/moderators":
-      presenceData.details = `Viewing Moderators`;
-      presenceData.state = `at Page ${
-        location.href.split("page=")[1].split("&")[0]
-      }`;
-      break;
-    case "/search":
-      presenceData.details = `Searching for ${escape(
-        location.href.split("term=")[1].split("&")[0]
-      )}`;
-      presenceData.state = `at Page ${
-        location.href.split("page=")[1].split("&")[0]
-      }`;
-      break;
-    default:
-      if (path.includes("cv")) {
-        const name = !path.endsWith("/rate")
-          ? document
-              .querySelector(
-                "body > div:nth-child(7) > div.user_box > div.container.left > div > h2"
-              )
-              ?.getAttribute("data-title")
-          : document
-              .querySelector("#box1 > div.vote_box > h1")
-              ?.textContent.split("#")[0];
-
-        presenceData.details = `${
-          path.endsWith("/rate") ? "Rating" : "Viewing"
-        } ${name || "Dummy"} CV`;
-        presenceData.buttons = [
-          {
-            url: location.href,
-            label: `Visit ${name || "Dummy"}'s CV`
-          }
-        ];
-      } else if (path.includes("u")) {
-        const name = document
-          .querySelector("body > div.profile_header > div > div > h1")
-          ?.textContent.split("#")[0];
-        presenceData.details = `Viewing ${name || "Dummy"} Profile`;
-        presenceData.buttons = [
-          {
-            url: location.href,
-            label: `Visit ${name || "Dummy"}'s Profile`
-          }
-        ];
-      } else presenceData.details = "Unknown Page";
-      break;
+if (document.location.pathname === "/") {
+  presenceData.details = "Viewing home page";
+} else if (document.location.pathname.includes("/search")) {
+  const search = document.location.href.split("term=")[1].split("&")[0],
+    page = document.location.href.split("page=")[1].split("&")[0];
+  presenceData.details = `🔍 Searching for: ${search || "Nothing"}`;
+  presenceData.state = `📖 Page ${page}`;
+  presenceData.buttons = [
+    {
+      label: "View Results",
+      url: document.location.href
+    }
+  ];
+} else if (document.location.pathname === "/moderators") {
+  const moderator_page = document.location.href
+      .split("page=")[1]
+      .split("&")[0],
+    filters = document.location.href.includes("&sort=");
+  presenceData.details = `Viewing 🔨 hireable moderators - Page: ${moderator_page}`;
```

Comment:
maybe this page number will be better in state?

Reply:
I tried that, not sure how to add it with the   if (filters) presenceData.state = `💿 Filters: True`; already at state..

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

Reply:
Oh true! Thanks. I have incorporated it.